### PR TITLE
Feature/filter

### PR DESCRIPTION
**Pull request by schittlur**

### CODE REVIEW

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 1]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+                const updatedParamObj = getUpdatedSearchQuery(location, paramObj, true);
+                const searchString = getSearchString(updatedParamObj);
+                history.push({
+                    pathname: '/search/',
+                    search: searchString
+                });
+            }          
+
+        }}>
+            <div data-testid="dropdownn"/>
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:363]: Two "n" characters?

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 2]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+            if (paramObj['sort'] === null){
+                const updatedParamObj = getUpdatedSearchQuery(location, paramObj, true);
+                const searchString = getSearchString(updatedParamObj);
+                history.push({
+                    pathname: '/search/',
+                    search: searchString + "&sort=" + event.value
+                });
+            }
+            else{
+                if (event.value === "MostRelavent"){
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:348]: "Relevant"

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 3]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+        for (let param in configuration.search_sort_options) {
+            options.push({value: configuration.search_sort_options[param].field_name, 
+                label: configuration.search_sort_options[param].display_name});
+        }
+    }
+    console.log(options);
+
+    let filterDropdown = (
+        <div data-testid='selectSort' id="sortDropdown">
+        <Select options={options} defaultValue={{label: "Most Relavent", value:"MostRelavent"}} name="sort" classNamePrefix='list'
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:332]: "Relevant"

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 4]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
         )
     }
 
+    let options = [{value:"MostRelavent", label: "Most Relavent"} ];
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:321]: "Relevant"

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 5]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+        }
+    }
+    console.log(options);
+
+    let filterDropdown = (
+        <div data-testid='selectSort' id="sortDropdown">
+        <Select options={options} defaultValue={{label: "Most Relavent", value:"MostRelavent"}} name="sort" classNamePrefix='list'  
+        data-testid="dropdown" placeholder="Select an option" className='dropdown'
+          onChange={event => {
+            // location = useLocation();
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:335]: Is this code pertinent?  If not it should be removed.

File: src/components/SearchResultsPage/SearchResultsPage.test.js [hunk 6]
```diff
@@ -181,4 +181,117 @@ describe('<SearchResultsPage />', () => {
         expect(await screen.findByText(errorText))
             .toBeInTheDocument();
     });
+
+    // it("Should display a functional dropdown for sort", async () => {
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.test.js:185]: Is the intention to merge all of this disabled code?

File: package.json [hunk 7]
```diff
@@ -11,10 +11,13 @@
     "query-string": "^6.14.0",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-dropdown": "^1.9.2",
```
Comment (jdiouf) [inline package.json:14]: Let's remove this if we're not using it anymore

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 8]
```diff
@@ -111,6 +116,27 @@ const SearchResultPage = (props) => {
         error: null
     });
 
+    const [sortState, setSortState] = useState({
```
Comment (jdiouf) [inline src/components/SearchResultsPage/SearchResultsPage.js:119]: We don';t need this state anymore I believe

File: src/components/SearchResultsPage/SearchResultsPage.test.js (hunk 6 above)
Comment (schittlur) [reply src/components/SearchResultsPage/SearchResultsPage.test.js:185]: yes

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 9]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
     }
 
+    let options = [{value:"MostRelavent", label: "Most Relavent"} ];
+    if(configuration!= null){
+        for (let param in configuration.search_sort_options) {
+            options.push({value: configuration.search_sort_options[param].field_name, 
+                label: configuration.search_sort_options[param].display_name});
+        }
+    }
+    console.log(options);
```
Comment (jdiouf) [inline src/components/SearchResultsPage/SearchResultsPage.js]: to be removed

**Review by mattdrayer — APPROVED** [review]

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 10]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+
+    let filterDropdown = (
+        <div data-testid='selectSort' id="sortDropdown">
+        <Select options={options} defaultValue={{label: "Most Relavent", value:"MostRelavent"}} name="sort" classNamePrefix='list'  
+        data-testid="dropdown" placeholder="Select an option" className='dropdown'
+          onChange={event => {
+            // location = useLocation();
+            let paramObj  = {};
+            console.log(event);
```
Comment (jdiouf) [inline src/components/SearchResultsPage/SearchResultsPage.js:335]: removing console log

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 11]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+            console.log(event.value);
```
Comment (jdiouf) [inline src/components/SearchResultsPage/SearchResultsPage.js:336]: removing console log

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 12]
```diff
@@ -291,6 +318,54 @@ const SearchResultPage = (props) => {
...
+            if (paramObj['sort'] === null){
+                const updatedParamObj = getUpdatedSearchQuery(location, paramObj, true);
+                const searchString = getSearchString(updatedParamObj);
+                history.push({
+                    pathname: '/search/',
+                    search: searchString + "&sort=" + event.value
```
Comment (jdiouf) [inline src/components/SearchResultsPage/SearchResultsPage.js:342]: Let's talk about what goes on in this block

**Review by jdiouf — APPROVED** [review]

**Pull request merged**